pygin/pygin/src/panel.cpp

```diff
@@ -0,0 +1,61 @@
+#include "headers.hpp"
+
+#include "panel.h"
+#include "far_api.hpp"
+#include "py.integer.hpp"
+
+PluginPanelItem*
+get_find_data::get_items(py::object const& find_data, size_t& item_count, OPERATION_MODES& mode)
+{
+	find_data.ensure_type(far_api::type("GetFindDataInfo"sv));
+	auto py_items = py::cast<py::list>(find_data["PanelItems"sv]);
+	auto size = py_items.size();
+	std::unique_ptr<char[]> mem(new char[sizeof(get_find_data)
+		+ sizeof(PluginPanelItem) * size]);
+	auto* result = reinterpret_cast<PluginPanelItem*>(
+		mem.get() + sizeof(get_find_data));
+	new (mem.get()) get_find_data(py_items, result);
+	item_count = size;
+	mode = py::cast<OPERATION_MODES>(find_data["OpMode"sv]);
+	mem.release();
+	return result;
+}
+
+void get_find_data::free(PluginPanelItem* items)
+{
+	delete reinterpret_cast<get_find_data*>(
```

Comment:
The pointer comes from `std::unique_ptr<char[]>`, but deleted using a scalar version of delete. This is UB.

If we're releasing certain smart pointers and expect to get them back from Far later, it's better  not to delete them manually, but wrap back into smart pointers of the same type and let their dtors do the job.

Reply:
Yeah, this code is wrong, good catch. I wonder why didn't it crash or something. The intention here is to call `get_find_data` destructor and then call delete char[].

Reply:
Fixed in 5ef1f612ff8787268b8d48658e3d1e8cad9b7272.